### ISSUE TITLE
Failing to read JSON files for any reason results in data loss

### ISSUE THREAD

**Issue by Funwayguy** [lorddusk/HQM#457]
This ties into the data loss side of #456 but is its own separate issue due to many other potential causes that could trigger it.

If for any reason the JSON files cannot be read due to access permissions or malformed due to an incorrect/incomplete write (forgetting to flush the buffers, concurrency bug, crash mid write, etc...), all subsequent data will also be lost where writes are still possible.

Even during a crash, the internal server still "shuts down" saving only what had been read thus and treats everything else as empty.

Suggestion: Have HQM set aside malformed JSON files before starting fresh and moving on with loading per-usual. This also helps with analysis of what went wrong and why without nuking the evidence.

**Comment by noobanidus** [comment]
Honestly, I don't understand the system well enough to cope with this. It's an unfortunately side effect and probably a good indicator as to why having regular backups is a thing. It's ideal, but I think it can wait until the rewrite.
